xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h

```diff
@@ -0,0 +1,385 @@
+ * You find the set of them on xbmc/addons/interfaces/General.cpp
+ *
+ * Note: For add-on development itself not needed, thats why with '*' here in
+ * text.
+ */
+typedef struct AddonToKodiFuncTable_kodi
+{
+  void (*open_settings_dialog)(void* kodiBase);
+  char* (*unknown_to_utf8)(void* kodiBase, const char* source, bool& ret, bool failOnBadChar);
+  char* (*get_localized_string)(void* kodiBase, long dwCode);
+  void (*get_dvd_menu_language)(void* kodiBase, char &language, unsigned int &iMaxStringSize);
+  void (*queue_notification)(void* kodiBase, const int type, const char* message);
+  void (*queue_notification_from_type)(void* kodiBase, const int type, const char* caption, const char* description, unsigned int displayTime, bool withSound, unsigned int messageTime);
+  void (*queue_notification_with_image)(void* kodiBase, const char* imageFile, const char* caption, const char* description, unsigned int displayTime, bool withSound, unsigned int messageTime);
+} AddonToKodiFuncTable_kodi;
+
+//==============================================================================
+/// \ingroup cpp_kodi_Defs
+/// @brief For kodi::QueueNotification() used message types
+///
+typedef enum QueueMsg
+{
+  /// Show info notification message
+  QUEUE_INFO,
+  /// Show warning notification message
+  QUEUE_WARNING,
+  /// Show error notification message
+  QUEUE_ERROR
+} QueueMsg;
+//------------------------------------------------------------------------------
+
+
+//==============================================================================
+namespace kodi {
+///
+/// \ingroup cpp_kodi
+/// @brief Opens this Add-Ons settings dialog.
+///
+///
+/// --------------------------------------------------------------------------
+///
+/// **Example:**
+/// ~~~~~~~~~~~~~{.cpp}
+/// #include <kodi/General.h>
+/// ..
+/// kodi::OpenSettings();
+/// ..
+/// ~~~~~~~~~~~~~
+///
+inline void OpenSettings()
+{
+  ::kodi::addon::CAddonBase::m_interface->toKodi->kodi->open_settings_dialog(::kodi::addon::CAddonBase::m_interface->toKodi->kodiBase);
+}
+} /* namespace kodi */
+//------------------------------------------------------------------------------
+
+//==============================================================================
+namespace kodi {
+///
+/// \ingroup cpp_kodi
+/// @brief Returns an addon's localized 'unicode string'.
+///
+/// @param[in] labelId    for string you want to localize.
+/// @param[in] defaultStr [opt] The default message, also helps to identify
+///                       the code that is used <em>(default is
+///                       <b><c>empty</c></b>)</em>
+/// @return               The localized message, or default if the add-on
+///                       helper fails to return a message
+///
+/// @note Label id's \b 30000 to \b 30999 and \b 32000 to \b 32999 are related
+/// to own add-on included strings inside
+/// <b>./resources/language/resource.language.??_??/strings.po</b>
+/// all others are from Kodi itself.
```

Comment:
" are related to the add-on's own included strings from ./resources/language/resource.language.??_??."
"All other strings are from Kodi core language files."